### PR TITLE
Export tier metrics

### PR DESCRIPTION
**Pull request by krisis**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
- `minio_node_tier_ttlb_seconds_distribution` - Distribution of time to last byte for streaming objects from warm tier
- `minio_node_tier_requests_success` - Number of requests to download object from warm tier that were successful 
- `minio_node_tier_requests_failure` - Number of requests to download object from warm tier that failed

## Motivation and Context
Observability into remote tier performance

## How to test this PR?
Sample output
```
./mc admin prometheus metrics myminio/ node | grep tier
# HELP minio_node_tier_requests_failure Number of requests to download object from warm tier that failed
# TYPE minio_node_tier_requests_failure counter
minio_node_tier_requests_failure{server="minio3:9000",tier="WARM-1"} 0
# HELP minio_node_tier_requests_success Number of requests to download object from warm tier that were successful
# TYPE minio_node_tier_requests_success counter
minio_node_tier_requests_success{server="minio3:9000",tier="WARM-1"} 1
# HELP minio_node_tier_ttlb_seconds_distribution Distribution of time to last byte for streaming objects from warm tier
# TYPE minio_node_tier_ttlb_seconds_distribution gauge
minio_node_tier_ttlb_seconds_distribution{le="0.010",server="minio3:9000",tier="WARM-1"} 0
minio_node_tier_ttlb_seconds_distribution{le="0.100",server="minio3:9000",tier="WARM-1"} 1
minio_node_tier_ttlb_seconds_distribution{le="1.000",server="minio3:9000",tier="WARM-1"} 1
minio_node_tier_ttlb_seconds_distribution{le="10.000",server="minio3:9000",tier="WARM-1"} 1
minio_node_tier_ttlb_seconds_distribution{le="2.000",server="minio3:9000",tier="WARM-1"} 1
minio_node_tier_ttlb_seconds_distribution{le="5.000",server="minio3:9000",tier="WARM-1"} 1

```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/tier.go [hunk 1]
```diff
@@ -80,6 +82,96 @@ type TierConfigMgr struct {
...
+
+var globalTierMetrics = tierMetrics{
+	requestsCount: make(map[string]struct {
+		success int64
+		failure int64
+	}),
+	histogram: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "tier_ttlb_seconds",
+		Help:    "Time taken by requests served by warm tier",
+		Buckets: []float64{0.01, 0.1, 1, 2, 5, 10},
```
Comment (vadmeste) [inline cmd/tier.go]: this bucket seems to be only useful for small files use case; maybe let's add more integer in the bucket for bigger files, I mean let's add 1m, 5m, 15m, 30m for more visibility to the user;

File: cmd/tier.go [hunk 2]
```diff
@@ -80,6 +82,96 @@ type TierConfigMgr struct {
...
+}
+
+var (
+	// {minio_node}_{tier}_{ttlb_seconds_distribution}
+	tierTTLBMD = MetricDescription{
+		Namespace: nodeMetricNamespace,
+		Subsystem: tierSubsystem,
+		Name:      ttlbDistribution,
+		Help:      "Distribution of time to last byte for objects downloaded from warm tier",
+		Type:      gaugeMetric,
```
Comment (vadmeste) [inline cmd/tier.go:135]: we are converting histogram to gauge here right?

Comment (krisis) [reply cmd/tier.go:135]: yes

File: cmd/tier.go (hunk 1 above)
Comment (krisis) [reply cmd/tier.go]: Good point, let me add those buckets too.

File: cmd/tier.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/tier.go:135]: I think this is handled manually via differently in our code, not sure why we have that kind of a code but this is apparently needed for now. 

Its the same problem with other TTFB values.

Comment (harshavardhana) [reply cmd/tier.go:135]: https://github.com/minio/minio/issues/17060 this issue is still open.

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**